src/components/organisms/SpaceCreateForm/SpaceCreateForm.tsx

```diff
@@ -0,0 +1,158 @@
+import React, { useCallback, useEffect, useMemo, useState } from "react";
+import { Form } from "react-bootstrap";
+import { useForm } from "react-hook-form";
+import { useHistory } from "react-router-dom";
+import { useAsyncFn } from "react-use";
+
+import { PortalInfoListItem, SPACE_INFO_LIST, SPACE_TAXON } from "settings";
+
+import { createSlug, createVenue_v2 } from "api/admin";
+
+import { spaceCreatePortalItem } from "store/actions/SpaceEdit";
+
+import { adminNGVenueUrl, generateAdminIaSpacePath } from "utils/url";
+import { buildEmptySpace } from "utils/venue";
+
+import { createSpaceSchema } from "forms/createSpaceSchema";
+
+import { useSpaceParams } from "hooks/spaces/useSpaceParams";
+import { useDispatch } from "hooks/useDispatch";
+import { useUser } from "hooks/useUser";
+
+import { AdminInput } from "components/molecules/AdminInput";
+import { AdminSection } from "components/molecules/AdminSection";
+import { FormCover } from "components/molecules/FormCover";
+import { FormErrors } from "components/molecules/FormErrors";
+import { PortalList } from "components/molecules/PortalList";
+import { SubmitError } from "components/molecules/SubmitError";
+import { YourUrlDisplay } from "components/molecules/YourUrlDisplay";
+
+import { ButtonNG } from "components/atoms/ButtonNG";
+
+import "./SpaceCreateForm.scss";
+
+// NOTE: add the keys of those errors that their respective fields have handled
+const HANDLED_ERRORS: string[] = ["venueName", "template"];
+
+export interface SpaceCreateFormProps {
+  worldId?: string;
+}
+
+export const SpaceCreateForm: React.FC<SpaceCreateFormProps> = ({
+  worldId,
+}) => {
+  const { worldSlug } = useSpaceParams();
+  const history = useHistory();
+  const dispatch = useDispatch();
+  const { user } = useUser();
+  const [selectedItem, setSelectedItem] = useState<
+    PortalInfoListItem | undefined
+  >();
+  const { icon: logoImageUrl, template } = selectedItem ?? {};
+
+  const { register, getValues, handleSubmit, errors, reset, watch } = useForm({
+    validationSchema: createSpaceSchema,
+    defaultValues: {
+      venueName: "",
+      template,
+    },
+  });
+
+  const [
+    { loading: isLoading, error: submitError },
+    addPortal,
```

Comment:
```suggestion
    createSpace,
```
We're creating spaces, not portals.

Reply:
renamed @ a7ee3f5a948443e1a5d242f317e57f69f49ba46c